### PR TITLE
chore(update blueprints): Updates contribution guidelines and code generator 

### LINKED ISSUE #499: Update contribution guidelines - standardize param construction
The param construction varies throughout components. 
Examples: 
`pf-c-alert--Modifier`
`pf-c-badge--modifier`
`btnClass`
`<div class="pf-c-content {{ modifierClass }}">`
`pf-c-dropdown--Modifier` 
`modal-box--modifiers`

The correct construction should be: `[partial]--modifiers` + `[partial]--attributes`

For example, in: `component.hbs`
```
<div class="classname{{#if component--modifier}} {{component--modifier}}{{/if}}"
  {{#if component--attribute}}
    {{{component--attribute}}}
  {{/if}}>
</div>

component--modifier relates to the partial itself, rather than the assigned class as the assigned class can change

```
Case for:
- `[component-name]--` rather than `pf-c-[component-name]--`
Using partial name (instead of classname) means that if a selector changes, you won't have to change every instance where that partial is called and modified (in example files). Also, the correct parameter name may not be immediately obvious from an example file. As a result, you have to reference the partial. A good example of this is: 
`{{#> card-body pf-c-card__content-modifier="" }}` vs `{{#> card-body card-body--modifier="" }}`

```
{{#> partial partial--modifier="[mod]" partial--attribute='id="[id]" partial--expandable="true" partial--expanded="true"}}
  content
{{/partial}}
```

rather than

```
{{#> partial pf-c-partial--modifiers="special" pf-c-partial--attributes='id="[id]" aria-label="[label]" aria-describedby="[description]"'  pf-c-partial--expandable="true" pf-c-partial--expanded="true"}}
  content
{{/partial}}
```

### LINKED ISSUE #500: Update component, layout, demo parameter definitions
Relates to #499 

Update component, layout, demo parameter definitions based on updated contribution guidelines.

### LINKED ISSUE #694: The blueprints have erros
The blueprints for the code generator have nested selectors and build error

### PR DESCRIPTION
**Pull request by mattnolting**
Closes #694 
Closes #500 
Closes #499

### CODE REVIEW

File: src/site/pages/contribution.md [hunk 1]
```diff
@@ -48,18 +48,21 @@ The main handlebars file for a block should be named using kebab case. For examp
...
+  {{#if card--attribute}}
+    {{{card--attribute}}}
+  {{/if}}>
   {{> @partial-block}}
 </div>
 ---
 <!-- Using the component in handlebars -->
-{{#> card pf-c-card--modifier="pf-m-3xl pf-m-margin"}}
-  {{> @partial-block}}
+{{#> card card--modifier="pf-m-3xl pf-m-margin" card--attribute='id="card-id" aria-lable="Card example"'}}
```
Comment (michael-coker) [inline src/site/pages/contribution.md]: `aria-lable` is missppelld :)

File: src/site/pages/contribution.md [hunk 2]
```diff
@@ -68,9 +71,9 @@ When including a partial within a partial, by default, Handlebars will pass alon
 If there is a possibility of a block nested inside another block of the same type and you want to isolate that nested block, add a new context. For example - see how the nested box is defined below with 'newcontext' added as an attribute:
 
 ```html
-{{#> card pf-c-card--modifier="pf-m-3xl pf-m-margin"}}
+{{#> card card--modifier="pf-m-3xl pf-m-margin" card--attribute='id="base-card" aria-label="Base card"'}}
```
Comment (michael-coker) [inline src/site/pages/contribution.md]: This is unrelated to the issue, but should we change this "card" example to another component? The modifiers used are for the "title" component, and `.pf-m-margin` doesn't exist anymore. Although I don't see another component would support nesting the component in itself. We could just remove `card--modifier` from these examples or use one of the layouts instead.

Comment (mattnolting) [reply src/site/pages/contribution.md]: Wellll, namespacing params resolves the issue w/nesting anyway. I'm going to remove it.

Comment (mattnolting) [reply src/site/pages/contribution.md]: @michael-coker maybe toolbar layout??

Comment (michael-coker) [reply src/site/pages/contribution.md]: Can you nest a toolbar in a toolbar? I didn't think so, but if you can, that works for me. You could nest a grid, split, or stack inside of itself.

**Pull request merged**